apps/common/controllers/add-another-address-loop.js

```diff
@@ -17,13 +17,11 @@ module.exports = class AddAnotherAddressController extends BaseController {
     const locals = super.locals(req, res);
     const addresses = req.sessionModel.get(`${this.field}Addresses`);
     const hasAddresses = req.sessionModel.get(`${this.field}Addresses`) ? true : false;
-    const items = _.map(addresses, (value, key) => ({
-      id: key,
-      address: value.address
-    }));
+    const items = this.mapAddress(addresses, req);
     return Object.assign({}, locals, {
       items,
-      hasAddresses
+      hasAddresses,
+      hasCategories: this.hasCategories(hasAddresses, addresses)
     });
```

Comment:
Better thanks @sulthan-ahmed. Are you going to move this into the parent class too?